src/object/srv_obj_migrate.c

```diff
@@ -475,6 +475,57 @@ migrate_pool_tls_lookup_create(struct ds_pool *pool, int version,
 	return tls;
 }
 
+static void
+mrone_recx_daos_vos_internal(struct migrate_one *mrone,
+			     struct daos_oclass_attr *oca,
+			     bool daos2vos, int shard)
+{
+	daos_iod_t *iod;
+	int cell_nr;
+	int stripe_nr;
+	int j;
+	int k;
+
+	D_ASSERT(DAOS_OC_IS_EC(oca));
+
+	cell_nr = obj_ec_cell_rec_nr(oca);
+	stripe_nr = obj_ec_stripe_rec_nr(oca);
+	/* Convert the DAOS to VOS EC offset */
+	for (j = 0; j < mrone->mo_iod_num; j++) {
+		iod = &mrone->mo_iods[j];
+		for (k = 0; k < iod->iod_nr; k++) {
+			daos_recx_t *recx;
+
+			recx = &iod->iod_recxs[k];
+			D_ASSERT(recx->rx_nr <= cell_nr);
+			if (daos2vos)
+				recx->rx_idx = obj_ec_idx_daos2vos(recx->rx_idx,
+								   stripe_nr,
+								   cell_nr);
+			else
+				recx->rx_idx = obj_ec_idx_vos2daos(recx->rx_idx,
+								   stripe_nr,
+								   cell_nr,
+								   shard);
```

Comment:
here it translates one recx between vos and daos, but a problem is that one continuous vos ext with rx_nr <= cell_nr possibly correspond to two daos recxs. 
For example 4+2 ec obj class, cell size 32, then one vos ext rx_idx = 10, rx_nr = 32 correspond to two daos recxs. not sure if it is a problem in the calling context here?

Reply:
oh, these extents are cell size boundary already after obj_recx_ec_daos2shard(). so it should not be a problem here. Though i probably should add assert here.